tests/test_polynomial.py

```diff
@@ -5,141 +5,90 @@
 
 class TestPolynomial(unittest.TestCase):
 
+    def setUp(self):
+        self.p1 = Polynomial(1, 2, 3)
+        self.p2 = Polynomial(3, 2, 1)
+        self.scalar = 5
+        self.zero_poly = Polynomial()
+
     def test_init(self):
-        # Test initialization from a sequence
-        p = Polynomial([1, 2, 3])
-        self.assertEqual(p.coeffs, [1, 2, 3])
+        # Test various ways of initializing a Polynomial
+        self.assertEqual(Polynomial(1, 2, 3).coeffs, [1, 2, 3])
+        self.assertEqual(Polynomial([1, 2, 3]).coeffs, [1, 2, 3])
+        self.assertEqual(Polynomial(self.p1).coeffs, self.p1.coeffs)
+        self.assertEqual(Polynomial(5).coeffs, [5])
+        self.assertEqual(Polynomial().coeffs, [])
+        self.assertEqual(Polynomial([1, 2, 3, 0, 0, 0]).coeffs, [1, 2, 3])
 
-        # Test initialization from scalars
```

Comment:
Why remove the comments related to specific way to initialize? I think that kind of reduces the value of comments, no?

Reply:
All initialization methods are well described in the definition of the `Polynomial` class constructor, so listing them again, in my opinion, does not make sense